### ISSUE TITLE
Issue with tungstenite 0.23.0

### ISSUE THREAD

**Issue by alcir** [desbma/gotify-desktop#19]
I was looking to package gotify-desktop in Feodra.
Since the version 0.23.0 of tungstenite will soon land in the Fedora repository, I was looking to update the dependency in Cargo.toml

```
tungstenite = { version = "0.23.0", default-features = false, features = ["native-tls", "handshake"] }
```

But trying to build gotify-desktop, I get

```
   Compiling gotify-desktop v1.3.6 (/home/alessio/rpmbuild/SPECS/gotify-desktop)
error[E0599]: no method named `into_client_request` found for struct `reqwest::Url` in the current scope
   --> src/gotify.rs:174:31
    |
174 |         let mut request = url.into_client_request()?;
    |                               ^^^^^^^^^^^^^^^^^^^ method not found in `Url`

warning: unused import: `client::IntoClientRequest`
  --> src/gotify.rs:14:19
   |
14 | use tungstenite::{client::IntoClientRequest, error::ProtocolError};
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

For more information about this error, try `rustc --explain E0599`.
warning: `gotify-desktop` (bin "gotify-desktop") generated 1 warning
error: could not compile `gotify-desktop` (bin "gotify-desktop") due to 1 previous error; 1 warning emitted
```
Any clue? Thanks.

**Comment by desbma** [comment]
I has a quick look, and migrating from tungstenite 0.20 to 0.23 requires some small changes.

I typically only update dependencies to new major version if there is a new feature I need, or known bug/CVE to fix.

I don't understand why you need to touch that for packaging though, since `gotify-desktop` is linked statically to its dependencies, including tungstenite? Does Fedora do something special when packaging Rust binaries?

**Comment by decathorpe** [comment]
We do statically link Rust (there's no sensible way to do dynamic linking after all). The only difference is that - in general - we don't use *vendored* dependencies. In this case, we already have updated our packages for the tungstenite and tokio-tungstenite crates to the latest versions to avoid having to provide packages for both v0.22 and v0.23, i.e. to match dependencies with the reqwest / rustls / tokio-rustls / hyper-rustls stack.

**Comment by desbma** [comment]
It should be fixed in 30a2a9f then.
